vertx-web-api-service/src/main/asciidoc/index.adoc

```diff
@@ -0,0 +1,185 @@
+
+[source,yaml]
+----
+/api/transactions:
+  get:
+    operationId: getTransactionsList
+    parameters:
+      - name: from
+        in: query
+        style: form
+        explode: false
+        required: false
+        schema:
+          type: string
+      - name: to
+        in: query
+        style: form
+        explode: false
+        required: false
+        schema:
+          type: string
+    responses: ...
+  put:
+    operationId: putTransaction
+    requestBody:
+      required: true
+      content:
+        application/json:
+          schema:
+            type: object
+            properties:
+              from:
+                type: string
+                format: email
+              to:
+                type: string
+                format: email
+              value:
+                type: number
+                format: double
+            additionalProperties: false
+            required:
+              - from
+              - to
+              - value
+    responses: ...
+----
+
+Now we can build the interface that specifies a *TransactionService* that handles _getTransactionsList_ and _putTransaction_ operations. For each operation you need to write a method with name corresponding to operation id with following parameters:
+
+* Last parameter must be a `Handler<AsyncResult<OperationResponse>>`
+* Second to last parameter must be a {@link io.vertx.ext.web.api.OperationRequest}
+* All parameters from first to second to last (excluded) are extracted from {@link io.vertx.ext.web.api.RequestParameter} with specified type automatically, but they need to respect https://vertx.io/docs/vertx-service-proxy/java/#_restrictions_for_service_interface[service proxy restrictions]
+
+For example:
+
+[source,java]
+----
+@WebApiServiceGen
+interface TransactionService {
+  void getTransactionsList(String from, String to, OperationRequest context, Handler<AsyncResult<OperationResponse>> resultHandler);
+  void putTransaction(JsonObject body, OperationRequest context, Handler<AsyncResult<OperationResponse>> resultHandler);
+}
+----
+
+When you receive a request at `TransactionService.getTransactionsList()` the generated service handler will automatically extract `from` and `to` parameter (if present) from {@link io.vertx.ext.web.api.OperationRequest}. In `putTransaction()` we use the `body` parameter name to extract the json body.
+
+The service handler is also capable to translate `JsonObject` to Vert.x data objects automatically, for example if you have a `Transaction` data object that matches the json schema above, you can rewrite the `putTransaction()` signature as:
+
+[source,java]
+----
+void putTransaction(Transaction body, OperationRequest context, Handler<AsyncResult<OperationResponse>> resultHandler);
+----
+
+Note that generated service handler matches method parameters with parameter name and not with the location of parameter, so avoid duplicated parameter names
```

Comment:
as far as I know, having two parameters with the same name would be a compilation issue, wouldn't it ?